components/DataBrowser/DataContainer.vue

```diff
@@ -58,105 +72,193 @@
 
 export default {
-  components: { SearchData, FilterData, DisplayData, Dashboard },
-  props: [ "category" ],
+  components: {
+    PaginationTool,
+    SearchData,
+    FilterData,
+    DisplayData,
+    Dashboard,
+  },
+  props: ["category"],
   data: () => {
     return {
-      isLoadingSearch: true,
-      totalCount: 0,
+      isLoading: true,
+      totalCount: undefined,
       currentData: [],
-      allFilterDict: {},
-      currentFilterDict: {},
-      file_type: [],
-      errorMessage: '',
-      searchContent: '',
-      relation: 'and',
-      sortBy: 'Published(asc)',
-    }
+      pageNumber: undefined,
+      limitNumber: undefined,
+      facetList: [],
+      filterDict: {},
+      searchContent: "",
+      relationType: true,
+      currentOrder: undefined,
+    };
   },
 
-  created: function () {
-    // when open find data page, call the function to fetch the data
-    this.dataChange(this.$route.query.type);
+  created: function() {
+    if (this.$route.query.facets) {
+      this.facetList = this.$route.query.facets.split(",");
+      this.relationType = this.$route.query.relation === "and" ? true : false;
+    } else {
+      this.fetchData();
+    }
+    if (this.$route.query.search) {
+      this.searchContent = this.$route.query.search;
+    }
+    if (this.$route.query.order) {
+      this.currentOrder = this.$route.query.order;
+    }
   },
 
   watch: {
-    // if the type variable in the url changes, change the current data to the data in that category
-    '$route.query.type': function (val) {
-      this.dataChange(val);
-    },
-
-    '$route.query.page': function () {
-      this.fetchData();
-    },
-
-    '$route.query.limit': function () {
-      this.fetchData();
+    "$route.query": {
+      handler() {
+        this.fetchData();
+      },
     },
   },
 
   methods: {
     async fetchData() {
-      this.isLoadingSearch = true;
-      let result = await backendQuery.fetchPaginationData(this.$config.query_api_url, this.currentFilterDict, this.$route.query.limit, this.$route.query.page, this.searchContent, this.relation, this.sortBy);
+      this.isLoading = true;
+      const result = await backendQuery.fetchPaginationData(
+        this.$config.query_api_url,
+        this.filterDict,
+        this.$route.query.limit,
+        this.$route.query.page,
+        this.searchContent,
+        this.relationType ? "and" : "or",
+        this.currentOrder
+      );
       this.currentData = result["items"];
       this.totalCount = result["total"];
-      this.isLoadingSearch = false;
+      this.isLoading = false;
     },
 
-    async dataChange(val) {
-      this.isLoadingSearch = true;
-      this.currentData = [];
-      if (val === 'dataset') {
-        this.allFilterDict = await backendQuery.fetchFilterData(this.$config.query_api_url, false);
+    compareFilter(oldFilter, newFilter) {
+      let isDifferent = false;
+      if (Object.keys(newFilter).length !== Object.keys(oldFilter).length) {
```

Comment:
Consider putting the lengths in a variable in this functions as the Object.keys are called in multiple places.